File: Week6/Problem/sentimental-readability/readability.py
def count_letters(text):
    # Compute the number of letters present in the text
    count_letters = 0

    for char in text:
        if char.isalpha():
            count_letters += 1

    return count_letters

def count_words(text):
    # Compute the number of words present in the text
    words = text.split()
    return len(words)

def count_sentences(text):
    # Compute the number of sentences in the text
    count_sentences = 0

    for char in text:
        if char in ['.', '!', '?']:
            count_sentences += 1

    return count_sentences

def get_grade(text):
    words = count_words(text)
    letters = count_letters(text)
    sentences = count_sentences(text)

    L_COEFF = 0.0588
    S_COEFF = 0.296
    COLEMAN_CONST = 15.8
    L = 0.0
    S = 0.0

    if words != 0:
        L = (letters / words) * 100
        S = (sentences / words) * 100

    index = (L_COEFF * L) - (S_COEFF * S) - COLEMAN_CONST

    return round(index)

if __name__ == "__main__":
    text = input("Text: ")
    grade = get_grade(text)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

File: Week2/Problem/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int get_grade(string text);

int main(void)
{
    string text = get_string("Text: ");
    int grade = get_grade(text);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    // compute the number of letters present in the text
    int countLetters = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            countLetters++;
        }
    }
    return countLetters;
}

int count_words(string text)
{
    // compute the number of word present in the text
    int countWords = 1;

    for (int i = 0; i < strlen(text); i++)
    {
        if (isblank(text[i]))
        {
            countWords++;
        }
    }

    return countWords;
}

int count_sentences(string text)
{
    // compute the number of sentences in the text
    int countSentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        char c = text[i];
        if (c == '.' || c == '!' || c == '?')
        {
            countSentences++;
        }
    }
    
    return countSentences;
}


int get_grade(string text)
{
    int words = count_words(text);
    int letters = count_letters(text);
    int sentences = count_sentences(text);

    float L_COEFF = 0.0588;
    float S_COEFF = 0.296;
    float COLEMAN_CONST = 15.8;
    float L = 0.0;
    float S = 0.0;

    if (words != 0)
    {
        L = (float) letters / words * 100;
        S = (float) sentences / words * 100;
    }

    float index = (L_COEFF * L) - (S_COEFF * S) - COLEMAN_CONST;

    return (int) round(index);
}